rust-lang/cargo
# refactor(registry): extract and rearrange items to their own modules

### What does this PR try to resolve?

A refactor moving things to their own new modules.

### How should we test and review this PR?

Commit by commit.

The PR just moves things around. I know it's a bit painful to see this kinda of change but to have a better module hierarchy we need to do it now or later.

### Additional information

This was done during coping with todos of asymmetric tokens #10519.
